nautobot/core/views/generic.py

```diff
@@ -74,6 +75,27 @@ def get_extra_context(self, request, instance):
         """
         return {}
 
+    def get_changelog_url(self, instance):
+        """Return the changelog URL for a given instance."""
+        meta = self.queryset.model._meta
+
+        # Don't try to generate a changelog_url for an ObjectChange.
+        if meta.model_name == "objectchange":
+            return None
+
+        route = f"{meta.app_label}:{meta.model_name}_changelog"
+
+        try:
+            changelog_url = reverse(route, kwargs={"pk": instance.pk})
+        except NoReverseMatch:
+            changelog_url = reverse(route, kwargs={"slug": instance.slug})
+        except Exception as err:
+            raise RuntimeError(
+                f"Unexpected error when retrieving changelog URL for object {self.queryset.model}: {err}"
+            )
```

Comment:
Is this correct/desirable behavior? Does it lead to 500 errors in the case of generic views for non-changelogged models? Would it be better/safer to return `None` instead of raising an error?

Reply:
This is just there for debug, thanks. I was just making sure that things were working. Although, I did not encounter a single detail view that didn't have change logs. As we discussed yesterday, best I could determine, these were the only objects that aren't change-logged, and not a single one of them have a detail view (`JobResult` instances being a special case):

```
 nautobot.circuits.models.CircuitTermination,
 nautobot.dcim.models.cables.CablePath,
 nautobot.dcim.models.device_component_templates.ComponentTemplateModel,
 nautobot.dcim.models.device_components.ComponentModel,
 nautobot.extras.models.change_logging.ObjectChange,
 nautobot.extras.models.customfields.CustomField,
 nautobot.extras.models.customfields.CustomFieldChoice,
 nautobot.extras.models.models.FileAttachment,
 nautobot.extras.models.models.FileProxy,
 nautobot.extras.models.models.ImageAttachment,
 nautobot.extras.models.models.JobResult,
 nautobot.extras.models.relationships.RelationshipAssociation,
 nautobot.extras.models.tags.TaggedItem,
 nautobot.users.models.ObjectPermission,
 nautobot.users.models.Token,
 nautobot.users.models.User,
```

In any case, I'll replace this with `None` and a sentinel check on whether to even render this nav tab just for completeness.

Reply:
Thanks! Note that a detail view for CustomField is coming in #735 so it'd be relevant then. :-)